### PR TITLE
Psse, add the following data blocks: switchedShuntData, TransformerImpedanceCorrectionTables, MultiSectionLineGrouping, InterareaTransferData and FactsDeviceData

### PR DESCRIPTION
**Pull request by marqueslanauja**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

Feature

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**
The following data blocks have been added to the import and export process:
- SwitchedShuntData
- TransformerImpedanceCorrectionTables
- MultiSectionLineGrouping
- InterareaTransferData
- FactsDeviceData

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: psse/psse-model/src/main/java/com/powsybl/psse/model/io/AbstractRecordGroupIOLegacyTextMultiLine.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/io/AbstractRecordGroupIOLegacyTextMultiLine.java:2]: *Copyright 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseFacts.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseFacts.java:2]: *Copyright 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseInterareaTransfer.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseInterareaTransfer.java:2]: *Copyright 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseLineGrouping.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseLineGrouping.java:2]: *Copyrighy 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PssePowerFlowModel.java [hunk 5]
```diff
@@ -105,6 +113,22 @@ public void addAreas(List<PsseArea> areas) {
         return Collections.unmodifiableList(areas);
     }
 
+    public void addTransformerImpedanceCorrections(List<PsseTransformerImpedanceCorrection> transformerImpedanceCorrections) {
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PssePowerFlowModel.java:116]: Should you check that the list is not `null`? Same thing for other methods in the file

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseTransformerImpedanceCorrection.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseTransformerImpedanceCorrection.java:2]: *Copyright 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseTransformerImpedanceCorrectionPoint.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseTransformerImpedanceCorrectionPoint.java:2]: *Copyright 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/FactsDeviceData.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/FactsDeviceData.java:2]: *Copyright 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/InterareaTransferData.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/InterareaTransferData.java:2]: *Copyright 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/MultiSectionLineGroupingData.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/MultiSectionLineGroupingData.java:2]: *Copyright 2021

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/TransformerImpedanceCorrectionTablesData.java [hunk 11]
```diff
@@ -0,0 +1,806 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
```
Comment (MioRtia) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/TransformerImpedanceCorrectionTablesData.java:2]: *Copyright 2021

**Review by MioRtia — COMMENTED** [review]
Small remarks. Most are to update the copyright

File: psse/psse-model/src/main/java/com/powsybl/psse/model/io/AbstractRecordGroupIOLegacyTextMultiLine.java (hunk 1 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/io/AbstractRecordGroupIOLegacyTextMultiLine.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseFacts.java (hunk 2 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseFacts.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseInterareaTransfer.java (hunk 3 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseInterareaTransfer.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseLineGrouping.java (hunk 4 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseLineGrouping.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseTransformerImpedanceCorrection.java (hunk 6 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseTransformerImpedanceCorrection.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseTransformerImpedanceCorrectionPoint.java (hunk 7 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseTransformerImpedanceCorrectionPoint.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/FactsDeviceData.java (hunk 8 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/FactsDeviceData.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/InterareaTransferData.java (hunk 9 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/InterareaTransferData.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/MultiSectionLineGroupingData.java (hunk 10 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/MultiSectionLineGroupingData.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/TransformerImpedanceCorrectionTablesData.java (hunk 11 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/TransformerImpedanceCorrectionTablesData.java]: Done

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PssePowerFlowModel.java (hunk 5 above)
Comment (marqueslanauja) [reply psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PssePowerFlowModel.java:116]: We believe that is not necessary as these methods are a simple call to the `adAll` method and the `adAll` method generates a `NullPointerException`  if the specified list is `null`

**Review by MioRtia — APPROVED** [review]

**Pull request merged**